scene/resources/material.cpp

```diff
@@ -1716,7 +1716,11 @@ void fragment() {)";
 		code += R"(
 	// Proximity Fade: Enabled
 	float proximity_depth_tex = textureLod(depth_texture, SCREEN_UV, 0.0).r;
+	#if CURRENT_RENDERER == RENDERER_COMPATIBILITY
+	vec4 proximity_view_pos = INV_PROJECTION_MATRIX * vec4(SCREEN_UV * 2.0 - 1.0, proximity_depth_tex * 2.0 - 1.0, 1.0);
```

Comment:
Alternately:
```suggestion
	vec4 proximity_view_pos = INV_PROJECTION_MATRIX * vec4(vec3(SCREEN_UV, proximity_depth_tex) * 2.0 - 1.0, 1.0);
```
We could also do something more like this, leaving the "default" case of Forward+/Mobile implicit:
```glsl
float proximity_depth_tex = textureLod(depth_texture, SCREEN_UV, 0.0).r;
#if CURRENT_RENDERER == RENDERER_COMPATIBILITY
proximity_depth_tex = proximity_depth_tex * 2.0 - 1.0;
#endif
vec4 proximity_view_pos = INV_PROJECTION_MATRIX * vec4(SCREEN_UV * 2.0 - 1.0, proximity_depth_tex, 1.0);
```

But I think the current form is the most explicit and clear about exactly what changes between the renderers.